cmd/geth/accountcmd.go

```diff
@@ -227,23 +232,46 @@ func accountList(ctx *cli.Context) error {
 }
 
 func accountProofOfPossession(ctx *cli.Context) error {
-	if len(ctx.Args()) != 2 {
+	if !ctx.IsSet(blsWalletFlag.Name) && len(ctx.Args()) != 2 {
 		utils.Fatalf("Please specify the address to prove possession of and the address to sign as proof-of-possession.")
 	}
+	if ctx.IsSet(blsWalletFlag.Name) && len(ctx.Args()) != 1 {
+		utils.Fatalf("Please specify the address to sign as proof-of-possession.")
+	}
 
 	stack, _ := makeConfigNode(ctx)
-	ks := stack.AccountManager().Backends(keystore.KeyStoreType)[0].(*keystore.KeyStore)
-
-	signer := common.HexToAddress(ctx.Args()[0])
-	message := common.HexToAddress(ctx.Args()[1])
-	account, _ := unlockAccount(ks, signer.String(), 0, utils.MakePasswordList(ctx))
+	am := stack.AccountManager()
+	ks := am.Backends(keystore.KeyStoreType)[0].(*keystore.KeyStore)
+	for _, wallet := range am.Wallets() {
```

Comment:
Looks like you are opening all the wallets.  Should only the wallet with the BLS address be opened?

Reply:
As far as I can tell you can't check this without opening the device first